fitbenchmarking/fitbenchmarking
# 391 exceptions

#### Description of Work

Fixes #391 

Implements a collection of custom exceptions and exception handler


#### Testing Instructions

1. Check exceptions are printed correctly

Function: Does the change do what it's supposed to?
Tests: Does it pass? Is there adequate coverage for new code?
Style: Is the coding style consistent? Is anything overly confusing?
Documentation: Is there a suitable change to documentation for this change?
